x-pack/platform/plugins/shared/onechat/public/application/hooks/tools/use_edit_tools.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EsqlToolDefinitionWithSchema, ToolDefinitionWithSchema } from '@kbn/onechat-common';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useCallback, useState } from 'react';
+import { UpdateToolPayload, UpdateToolResponse } from '../../../../common/http_api/tools';
+import { queryKeys } from '../../query_keys';
+import { useOnechatServices } from '../use_onechat_service';
+
+export const useEditTool = ({
+  onSuccess,
+  onError,
+}: {
+  onSuccess?: (tool: UpdateToolResponse) => void;
+  onError?: (error: Error) => void;
+}) => {
+  const queryClient = useQueryClient();
+  const { toolsService } = useOnechatServices();
+
+  const { mutateAsync, isLoading } = useMutation({
+    mutationFn: ({ toolId, tool }: { toolId: string; tool: UpdateToolPayload }) =>
+      toolsService.update(toolId, tool),
+    onSuccess,
+    onError,
+    onSettled: () => queryClient.invalidateQueries({ queryKey: queryKeys.tools.all }),
```

Comment:
(related to previous comment) if we access the fetch the tool from the flyout, we gonna need to invalidate the tool's specific key/query too (which we don't have a definition for atm)